src/emulator/storage/files.ts

```diff
@@ -331,6 +331,37 @@ export class StorageLayer {
     return metadata;
   }
 
+  public async checkResumableUploadAuthorization(upload: Upload): Promise<boolean> {
+    const storedMetadata = this.getMetadata(upload.bucketId, upload.objectId);
+    const metadata = new StoredFileMetadata(
+      {
+        name: upload.objectId,
+        bucket: upload.bucketId,
+        contentType: upload.metadata.contentType || "application/octet-stream",
+        contentDisposition: upload.metadata.contentDisposition,
+        contentEncoding: upload.metadata.contentEncoding,
+        contentLanguage: upload.metadata.contentLanguage,
+        cacheControl: upload.metadata.cacheControl,
+        customMetadata: upload.metadata.metadata,
+      },
+      this._cloudFunctions,
+      Buffer.alloc(0)
```

Comment:
This is potentially going to give us incorrect security rule results if there's a security rule that checks the `request.size` variable.

This is also essentially why we wait until the file is fully uploaded before evaluating security rules, because we need an accurate file size before we can get a fully formed rules verdict.

Reply:
e.g. the following security rule with _always_ fail given the above logic

```
allow write: if request.resource.size > 0;
```